clang/include/clang/AST/TypeProperties.td

```diff
@@ -668,12 +668,16 @@ let Class = AttributedType in {
   def : Property<"equivalentType", QualType> {
     let Read = [{ node->getEquivalentType() }];
   }
-  def : Property<"attribute", AttrKind> {
+  def : Property<"attrKind", AttrKind> {
     let Read = [{ node->getAttrKind() }];
   }
+  def : Property<"attribute", Attr> {
```

Comment:
Is there reason to store both `attribute` and `attrKind` here?  Should we just implement `attrKind` as `attribute.getKind` ?

Reply:
Yes, Attr is not always available (nullability related attributes only have a kind and don't allocate Attr).